onedal/datatypes/_data_conversion.py

```diff
@@ -21,39 +21,79 @@
 from daal4py.sklearn._utils import make2d
 from onedal import _backend, _is_dpc_backend
 
+from .._device_offload import dpctl_available, dpnp_available
 from ..utils import _is_csr
 
-try:
-    import dpctl
+if dpctl_available:
     import dpctl.tensor as dpt
 
-    dpctl_available = dpctl.__version__ >= "0.14"
-except ImportError:
-    dpctl_available = False
+if dpnp_available:
+    import dpnp
 
 
-def _apply_and_pass(func, *args):
+def _apply_and_pass(func, *args, **kwargs):
     if len(args) == 1:
-        return func(args[0])
-    return tuple(map(func, args))
-
-
-def from_table(*args):
-    return _apply_and_pass(_backend.from_table, *args)
-
-
-def convert_one_to_table(arg):
-    if dpctl_available:
-        if isinstance(arg, dpt.usm_ndarray):
-            return _backend.dpctl_to_table(arg)
+        return func(args[0], **kwargs) if len(kwargs) > 0 else func(args[0])
+    return (
+        tuple(func(arg, **kwargs) for arg in args)
+        if len(kwargs) > 0
+        else tuple(func(arg) for arg in args)
+    )
```

Comment:
```suggestion
        return func(args[0], **kwargs)
    return tuple(map(lambda arg: func(arg, **kwargs), args))
    )
```
Does a lambda accomplish the necessary changes here? Or could you list a circumstance where a zero-length kwarg will throw things off?

Reply:
Thank you @icfaust ! for the suggestion. It properly works.
My previous tries used to have some problems with the zero length of kwargs, so lambda really accomplish the necessary changes.